timesync/clock.go

```diff
@@ -114,9 +124,21 @@ func (t *NodeClock) tick() {
 	if t.clock.Now().Before(t.genesis) {
 		return
 	}
-
 	layer := t.TimeToLayer(t.clock.Now())
 
+	// why it fails in tests?
+	if !layer.Before(t.lastTickedLayer) {
+		d := layer.Difference(t.lastTickedLayer)
+		if d > 1 {
+			t.log.Warning("tick is too late",
+				log.Stringer("current", layer),
+				log.Stringer("last", t.lastTickedLayer),
+			)
+		}
+		if d != 0 {
+			tickDistance.Observe(float64(d))
+		}
+	}
```

Comment:
I am confused by the log. If `layer` is before `t.lastTickedLayer` the clock moved backwards. The tick isn't to late (a tick is once every second) but rather the clock was 12:00:00 at some point, starting a new layer and then moved back to e.g. 11:59:00 and is now 2 layers prior the one it already was in.

The tick isn't to late, the clock just moved backwards in time.